miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppDownloader.kt

```diff
@@ -20,23 +20,31 @@ internal class MiniAppDownloader(
 ) : UpdatableApiClient {
 
     // Only run the latest version of specified MiniApp.
-    suspend fun getMiniApp(appId: String, versionId: String): String = when {
-        !isLatestVersion(appId, versionId) -> throw sdkExceptionForInvalidVersion()
-        miniAppStatus.isVersionDownloaded(appId, versionId, storage.getMiniAppVersionPath(appId, versionId))
-        -> storage.getMiniAppVersionPath(appId, versionId)
-        else -> startDownload(appId, versionId)
-    }
-
-    @Suppress("TooGenericExceptionCaught", "SwallowedException")
-    private suspend fun isLatestVersion(appId: String, versionId: String): Boolean {
+    @Suppress("SwallowedException", "LongMethod")
+    suspend fun getMiniApp(appId: String, versionId: String): String {
         try {
-            return apiClient.fetchInfo(appId).version.versionId == versionId
-        } catch (e: Exception) {
-            // If backend functions correctly, this should never happen
-            throw sdkExceptionForInternalServerError()
+            return when {
+                !isLatestVersion(appId, versionId) -> throw sdkExceptionForInvalidVersion()
+                miniAppStatus.isVersionDownloaded(
+                    appId,
+                    versionId,
+                    storage.getMiniAppVersionPath(appId, versionId)
```

Comment:
The version path is used several times here. Maybe you can set it in a `val versionPath` before the `try/catch`?